src/KissThreading.jl

```diff
@@ -141,11 +142,13 @@ end
     tmapreduce(f, op, src::AbstractArray...; init, batch_size=default_batch_size(length(src[1])))
 
 $(_doc_threaded_version(mapreduce))
+
+Warning: In contrast to `Base.mapreduce` it is assumed that `op` must be commutative. Otherwise
```

Comment:
No the semantics of `mapreduce` is only associativity of `op`. Order as in `foldl` is unrealated to commutativity, but related to associativity:
Given `[x1, x2, x3]` foldl guarantees the computation `(x1*x2)*x3`. Mapreduce says it can be either `(x1*x2)*x3` or `x1*(x2*x3)`, but not e.g. `(x1 * x3) * x2`.

Reply:
It is in fact somewhat common to use `reduce` in situations, where operands even live in different spaces, so switching them would error :
```julia
julia> reduce(push!, 1:5, init=Int[])
5-element Array{Int64,1}:
 1
 2
 3
 4
 5
```